pkg/reconciler/pipeline/validate.go

```diff
@@ -89,6 +89,17 @@ func ValidatePipeline(pl *dfv1.Pipeline) error {
 		return fmt.Errorf("pipeline has no sink, at least one vertex with 'sink' defined is required")
 	}
 
+	for _, v := range pl.Spec.Vertices {
+		if len(pl.GetToEdges(v.Name)) == 0 {
+			if len(pl.GetFromEdges(v.Name)) == 0 {
+				return fmt.Errorf("not all the vertex names are defined in edges")
```

Comment:
Is this error message right?

Reply:
This is the error message on lines 181-183 that handles dangling vertices as well.

I think a more appropriate message might be that the vertex is not connected to the pipeline.

Reply:
yes, would be better if we embed the vertex name into the err msg.

Reply:
This line may not be necessary now with this implementation. Since we are checking that each UDF/Source/Sink has the appropriate number of edges, those error messages will inform us which vertex is dangling.